src/datadog/baggage.cpp

```diff
@@ -0,0 +1,177 @@
+#include <datadog/baggage.h>
+
+namespace datadog {
+namespace tracing {
+
+namespace {
+
+Expected<std::unordered_map<std::string, std::string>, Baggage::Error>
+parse_baggage(StringView input, size_t max_capacity) {
+  std::unordered_map<std::string, std::string> result;
+  if (input.empty()) return result;
+
+  enum class state : char {
+    leading_spaces_key,
+    key,
+    leading_spaces_value,
+    value
+  } internal_state = state::leading_spaces_key;
+
+  size_t beg = 0;
+  size_t tmp_end = 0;
+
+  StringView key;
+  StringView value;
+
+  const size_t end = input.size();
+
+  for (size_t i = 0; i < end; ++i) {
+    switch (internal_state) {
+      case state::leading_spaces_key: {
+        if (input[i] != ' ') {
```

Comment:
I think we also need to check for `\t`

```suggestion
        if (input[i] != ' ' && input[i] != '\t') {
```

Reply:
And I think this can also be applied elsewhere. I'm not sure how efficient it is, but maybe you could define a lambda expression in this function that evaluates whether a char is optional whitespace

Reply:
Adressed in e3394c9f9e776b0f769c160914b30aeda6dea13a